marks_reader_python/marks_reader.py

```
@@ -0,0 +1,436 @@
+    csv_attendance = os.path.join(output_dir, f"{class_code}_attendance.csv")
+    csv_transpose = os.path.join(output_dir, f"{class_code}_marks_transposed.csv")
+
+    # Read all students
+    students = []
+    with open(rec_file, 'rb') as f:
+        while True:
+            try:
+                student = read_student_record(f)
+                if student['name']:  # Skip empty records
+                    students.append(student)
+            except:
```

Ruff (0.13.3)

Do not use bare `except`

(E722)